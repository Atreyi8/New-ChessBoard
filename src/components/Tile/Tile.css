.tile {
  position: relative;
  width: 60px;
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.hover-label {
  position: absolute;
  bottom: 4px;
  right: 4px;
  background-color: rgba(0, 0, 0, 0.5);
  color: white;
  font-size: 12px;
  padding: 2px 4px;
  border-radius: 4px;
  display: none; 
  pointer-events: none;
}


.tile:hover .hover-label,
.selected-tile .hover-label {
  display: block;
}

.selected-tile {
  border: 3px solid yellow;
  box-sizing: border-box;
}

.white-tile {
  background-color: #b2afaf;
}

.black-tile {
  background-color: #555;
}

.valid-move {
  background-color: lightgreen; 
}