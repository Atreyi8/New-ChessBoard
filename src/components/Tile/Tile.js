import React from "react";
import Piece from "../Piece/Piece";
import "./Tile.css";

const Tile = ({ row, col, tileData, onClick, isSelected }) => {
  const tileClass = (row + col) % 2 === 0 ? "white-tile" : "black-tile";
  const selectedClass = isSelected ? "selected-tile" : "";

  return (
    <div
      className={`tile ${tileClass} ${selectedClass}`}
      onClick={onClick}
      data-row={8 - row} // row numbers starts from 8 at the top
      data-col={String.fromCharCode(65 + col)} // A-H cpls
    >
      {tileData && tileData.symbol && (
        <Piece symbol={tileData.symbol} color={tileData.color} />
      )}
      <div className="hover-label">
        {String.fromCharCode(65 + col)}{8 - row}
      </div>
    </div>
  );
};

export default Tile;
