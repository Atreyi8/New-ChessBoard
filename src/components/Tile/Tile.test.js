import React from "react";
import { render } from "@testing-library/react";
import Tile from "./Tile";

describe("Tile Component Snapshot", () => {
  const defaultProps = {
    row: 0,
    col: 0,
    tileData: { symbol: "P", color: "white" },
    onClick: jest.fn(),
    isSelected: false,
  };

  it("matches snapshot for white-tile", () => {
    const { asFragment } = render(<Tile {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for black-tile", () => {
    const { asFragment } = render(<Tile {...defaultProps} col={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when selected", () => {
    const { asFragment } = render(<Tile {...defaultProps} isSelected />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with custom hover label", () => {
    const { asFragment } = render(<Tile {...defaultProps} row={2} col={3} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
