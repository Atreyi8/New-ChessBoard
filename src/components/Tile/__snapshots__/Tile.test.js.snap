// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Component Snapshot matches snapshot for black-tile 1`] = `
<DocumentFragment>
  <div
    class="tile black-tile "
    data-col="B"
    data-row="8"
  >
    <span
      class="piece white"
    >
      P
    </span>
    <div
      class="hover-label"
    >
      B8
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tile Component Snapshot matches snapshot for white-tile 1`] = `
<DocumentFragment>
  <div
    class="tile white-tile "
    data-col="A"
    data-row="8"
  >
    <span
      class="piece white"
    >
      P
    </span>
    <div
      class="hover-label"
    >
      A8
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tile Component Snapshot matches snapshot when selected 1`] = `
<DocumentFragment>
  <div
    class="tile white-tile selected-tile"
    data-col="A"
    data-row="8"
  >
    <span
      class="piece white"
    >
      P
    </span>
    <div
      class="hover-label"
    >
      A8
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tile Component Snapshot matches snapshot with custom hover label 1`] = `
<DocumentFragment>
  <div
    class="tile black-tile "
    data-col="D"
    data-row="6"
  >
    <span
      class="piece white"
    >
      P
    </span>
    <div
      class="hover-label"
    >
      D6
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tile Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="tile white-tile "
    data-col="B"
    data-row="7"
  >
    <div
      data-testid="piece"
    >
      P (white)
    </div>
    <div
      class="hover-label"
    >
      B7
    </div>
  </div>
</DocumentFragment>
`;
